### ISSUE TITLE
[Bug]: Paint events do not fire when window is initially hidden on Windows or Linux

### ISSUE THREAD

**Issue by mitchchn** [electron/electron#32001]
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a feature request that matches the one I want to file, without success.

### Electron Version

16.0.0

### What operating system are you using?

Other (specify below)

### Operating System Version

Windows 11, macOS 12, Fedora 35

### What arch are you using?

Other (specify below)

### Last Known Working Electron version

11.5.0

### Expected Behavior

A hidden window with this configuration should still paint:

```ts
  const mainWindow = new BrowserWindow({
    width: 800,
    height: 600,
    show: false,
    paintWhenInitiallyHidden: true,
  })
```

The page should fire paint events that can be picked up by `PerformanceObserver`.

```ts
const observer = new PerformanceObserver((performance) => {
  console.log("Paint event:", performance.getEntries(), new Date().toString());
});
observer.observe({ entryTypes: ["paint"] });
```

### Actual Behavior

On Windows and Linux, paint events do not fire while the window is hidden. It is unclear whether any painting is happening while the window is hidden.

On macOS, it works as expected.

Windows console output in renderer:

````
Contentful element added to DOM: <h1>​Paint me!!​</h1>​
Showing window after timeout: Thu Nov 25 2021 13:26:28 GMT-0500 (Eastern Standard Time)
Paint event: Array(2) Thu Nov 25 2021 13:26:28 GMT-0500 (Eastern Standard Time)
````

Mac console output in renderer:

```
Contentful element added to DOM: <h1>​Paint me!!​</h1>​
Paint event: Array(2) Thu Nov 25 2021 13:25:10 GMT-0500 (Eastern Standard Time)
Showing window after timeout: Thu Nov 25 2021 13:25:13 GMT-0500 (Eastern Standard Time)
````

### Testcase Gist URL

https://gist.github.com/062c602a24a5f72a1f00d05285fdac68

### Additional Information

This behaviour seemed to regress gradually over the course of several Electron versions. In 11.x, the paint events reliably fire when the window is hidden. Around 13-14, they sometimes do, and sometimes do not. As of 16.0, they never fire on Windows or Linux, but continue to fire on macOS.

**Comment by pushkin-** [comment]
I think this is still an issue in Electron 21

**Comment by fanchenio** [comment]
Indeed, for example, when I want to hide the window, I want to clear the content of the input box on the html, but when the window is hidden, after sending a message to the page through contents.send, it will not take effect immediately, but will take effect after the window is displayed.

**Comment by pushkin-** [comment]
bump

**Comment by pushkin-** [comment]
repros in v26

**Comment by rudyrichter** [comment]
In v35 this seems to also be impacting macOS as well.

**Comment by nikwen** [comment]
I ran into this while working on https://github.com/electron/electron/pull/49706.
